src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -2511,20 +2511,231 @@ TEST_CASE("settings upgrade command line utils", "[tx][soroban][upgrades]")
     auto const& contractSourceRefLedgerKey = std::get<1>(createRes);
     auto const& contractID = std::get<2>(createRes);
 
+    xdr::xvector<ConfigSettingEntry> initialEntries;
     xdr::xvector<ConfigSettingEntry> updatedEntries;
     for (uint32_t i = 0;
          i < static_cast<uint32_t>(CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW); ++i)
     {
         LedgerTxn ltx(app->getLedgerTxnRoot());
         auto entry =
             ltx.load(configSettingKey(static_cast<ConfigSettingID>(i)));
+
+        // Store the initial entries before we modify the cost types below
+        initialEntries.emplace_back(entry.current().data.configSetting());
+
+        // We're writing the latest cpu and mem settings below to makes sure
+        // they don't brick the settings upgrade process. We would ideally pull
+        // this from phase1.json, but core doesn't know how to parse JSON XDR.
+        auto const& vals = xdr::xdr_traits<ContractCostType>::enum_values();
         if (entry.current().data.configSetting().configSettingID() ==
-            CONFIG_SETTING_CONTRACT_LEDGER_COST_V0)
+            CONFIG_SETTING_CONTRACT_COST_PARAMS_CPU_INSTRUCTIONS)
         {
-            entry.current()
-                .data.configSetting()
-                .contractLedgerCost()
-                .feeRead1KB = 1234;
+            auto& params = entry.current()
+                               .data.configSetting()
+                               .contractCostParamsCpuInsns();
+            for (auto val : vals)
```

Comment:
Instead of hardcoding the values here we should probably read the json instead. This is both less cumbersome and more safe. Fine to do as a followup though.